src/main/java/de/tum/in/www1/artemis/web/rest/UserResource.java

```diff
@@ -225,4 +225,41 @@ public ResponseEntity<Void> deleteSshPublicKey() {
         log.debug("Successfully deleted SSH key of user {}", user.getLogin());
         return ResponseEntity.ok().build();
     }
+
+    /**
+     * PUT users/vcsAccessToken : creates a vcsAccessToken for a user
+     *
+     * @return the ResponseEntity with a userDTO containing the token: with status 200 (OK), with status 404 (Not Found), or with status 400 (Bad Request)
+     */
+    @PutMapping("users/vcsAccessToken")
+    @EnforceAtLeastStudent
+    public ResponseEntity<UserDTO> createVcsAccessToken() {
+        User user = userRepository.getUser();
+        log.debug("REST request to create a new VCS access token for user {}", user.getLogin());
+
+        // TODO add token generation here -> depends on https://github.com/ls1intum/Artemis/pull/8929
```

Comment:
Please make the time it is valid configurable :)

Reply:
Added a date picker, so it can be configurable :+1: